### ISSUE TITLE
Implement step "out"

### ISSUE THREAD

**Issue by radare** [radare/radare2#4708]
Looks like Microsoft describes `step out` as a way to continue until end of function. we already have continue until-ret, but this is not really correct. so we must have a way to continue execution until the function is returning.

This is arch-specific, but it basically requires grabbing the return address from the stack or the register in case the arch uses that.

**Comment by jduck** [comment]
I think both features are useful, but they both need a test case in R2R. I can take a stab Sunday or next week if no one else gets to it by then.

**Comment by Maijin** [comment]
@leberus ^

**Comment by leberus** [comment]
Is this not the same as return until ret? I mean, "step out" should put a breakpoint in the ret instruction itself, or once we returned from the function?

**Comment by radare** [comment]
No. A step until ret continues the exec until the next ret. but this will read the return address from the backtrace info. Aka dbt, this will continue until that address is hit. This is how frida traces work and its imho the best way to find the exitpoints of a function

**Comment by radare** [comment]
I mean .. we can do that by hand, but its always handy to have a command for it

**Comment by ret2libc** [comment]
This issue has been moved from radareorg/radare2 to radareorg/ideas as we are trying to clean our backlog and this issue has probably been created a long while ago. This is an effort to help contributors understand what are the actionable items they can work on, prioritize issues better and help users find active/duplicated issues more easily. If this is not an enhancement/improvement/general idea but a bug, feel free to ask for re-transfer to main repo. Thanks for your understanding and contribution with this issue.
